sycl/test-e2e/Annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,121 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// E2E test for annotated USM allocation
+
+#include "sycl/sycl.hpp"
+#include <iostream>
+
+// clang-format on
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+
+using alloc = sycl::usm::alloc;
+
+constexpr int N = 10;
+
+void TestUsmKind(sycl::queue &q) {
+  const sycl::context &Ctx = q.get_context();
+  auto dev = q.get_device();
+
+  // the USM kind is specified in the function name
+  properties P1{conduit, stable};
+  auto APtr1 = malloc_device_annotated<int>(N, q, P1);
+  assert(sycl::get_pointer_type(APtr1.get(), Ctx) == alloc::device);
+
+  auto APtr2 = aligned_alloc_device_annotated(128, N, q);
+  assert(sycl::get_pointer_type(APtr2.get(), Ctx) == alloc::device);
+
+  auto APtr3 = malloc_host_annotated<int>(N, q);
+  assert(sycl::get_pointer_type(APtr3.get(), Ctx) == alloc::host);
+
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    auto APtr4 = malloc_shared_annotated(N, q);
+    assert(sycl::get_pointer_type(APtr4.get(), Ctx) == alloc::shared);
+    free(APtr4, q);
+  }
+
+  // the USM kind is specified in the propList
+  properties P2{conduit, cache_config{large_slm}};
+  auto APtr5 = malloc_annotated<int>(N, q, alloc::device, P2);
+  assert(sycl::get_pointer_type(APtr5.get(), Ctx) == alloc::device);
+
+  // the USM kind is specified in the propList
```

Comment:
Can you add a negative test where the function requires the usm_kind property, but the properties argument does not contain usm_kind? This should produce a nice error message.